streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -227,6 +255,8 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
       case DeregisterReceiver(streamId, message, error) =>
         deregisterReceiver(streamId, message, error)
         context.reply(true)
+      case ScheduleReceiver(streamId) =>
```

Comment:
Sounds great. One potential issue is if we reject a receiver using `!scheduleReceiver(receiverId).contains(host))` when receiving `RegisterReceiver`, we may keep restarting a receiver if the scheduled executors keep busy, such as running some long running jobs.

My current implementation only rejects a receiver for a mismatch location when restarting it.